x-pack/plugins/security/server/lib/cluster_security_features/cluster_security_features.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  XPackUsageContract,
+  XPackUsageResponse,
+  SecurityRealm,
+} from '../../../../xpack_main/server/lib/xpack_usage';
+
+export class ClusterSecurityFeatures {
+  private currentXPackUsage: XPackUsageResponse | undefined;
+
+  private enabledRealms: SecurityRealm[] | undefined;
+
+  constructor(xpackUsage: XPackUsageContract) {
+    xpackUsage.getUsage$().subscribe({
+      next: updatedUsage => this.updateCurrentUsage(updatedUsage),
+    });
+  }
+
+  public isSAMLRealmEnabled(): boolean {
+    return this.enabledRealms ? this.enabledRealms.includes('saml') : false;
```

Comment:
We could optionally "fail open" here. If we don't have any realm information, then assume that SAML is enabled, and then see if ES happens to reject the call.